aspnetcore/fundamentals/target-aspnetcore.md

```diff
@@ -0,0 +1,213 @@
+
+### Razor views or Razor Pages
+
+A project that includes [Razor views](xref:mvc/views/overview) or [Razor Pages](xref:razor-pages/index) must use the [Microsoft.NET.Sdk.Razor SDK](xref:razor-pages/sdk).
+
+If the project targets .NET Core 3.0, it requires:
+
+* An `AddRazorSupportForMvc` MSBuild property set to `true`.
+* A `<FrameworkReference>` element for the shared framework.
+
+For example:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netcoreapp3.0-razor-views-pages-library.csproj)]
+
+If the project targets .NET Standard instead, it requires a [Microsoft.AspNetCore.Mvc](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc) package reference. The `Microsoft.AspNetCore.Mvc` package moved into the shared framework and is therefore no longer published. For example:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netstandard2.0-razor-views-pages-library.csproj?highlight=8)]
+
+### Razor components
+
+A project that includes [Razor components](xref:blazor/components) must use the [Microsoft.NET.Sdk.Razor SDK](xref:razor-pages/sdk).
+
+To support Razor component consumption from [Blazor Server](xref:blazor/hosting-models#blazor-server) only:
+
+* Target .NET Core 3.0.
+* Add a `<FrameworkReference>` element for the shared framework.
+
+For example:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netcoreapp3.0-razor-components-library.csproj)]
+
+To support Razor component consumption from both Blazor Server and [Blazor WebAssembly](xref:blazor/hosting-models#blazor-webassembly):
+
+* Target .NET Standard 2.0.
+* Add the following package references:
+  * [Microsoft.AspNetCore.Components](https://www.nuget.org/packages/Microsoft.AspNetCore.Components)
+  * [Microsoft.AspNetCore.Components.Web](https://www.nuget.org/packages/Microsoft.AspNetCore.Components.Web)
+* Set the `RazorLangVersion` property to `3.0`. `3.0` is the default value for .NET Core 3.0.
+
+For example:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netstandard2.0-razor-components-library.csproj)]
+
+For more information on libraries containing Razor components, see [ASP.NET Core Razor components class libraries](xref:blazor/class-libraries).
+
+### Tag Helpers
+
+A project that includes [Tag Helpers](xref:mvc/views/tag-helpers/intro) should use the `Microsoft.NET.Sdk` SDK. If targeting .NET Core 3.0, add a `<FrameworkReference>` element for the shared framework. For example:
+
+[!code-xml[](target-aspnetcore/samples/single-tfm/netcoreapp3.0-basic-library.csproj)]
+
+If targeting .NET Standard, add a package reference to [Microsoft.AspNetCore.Mvc.Razor](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor). The `Microsoft.AspNetCore.Mvc.Razor` package moved into the shared framework and is therefore no longer published. For example:
```

Comment:
The statement "The `Microsoft.AspNetCore.Mvc.Razor` package moved into the shared framework and is therefore no longer published." could be confusing here. What is the thinking behind mentioning it again? Maybe whenever we say "If targeting .NET Standard," we should extend it to clarify the *why*, e.g. "If targeting .NET Standard (in order to provide support for versions earlier than ASP.NET Core 3.0).."